### ISSUE TITLE
Invalid arm64 executable for Apple M1

### ISSUE THREAD

**Issue by bjbishop** [mmmorris1975/aws-runas#67]
Hi Mike!  Thanks for making this great tool; use it daily and we're forever in your debt!  :)

I'm lucky enough to have just received a fancy new Macbook Pro with the M1 processor and trying to install aws-runas 3.1.2 darwin-arm64.zip and it appears that the binary is invalid?  Downloading the source and compiling on my mac produced a valid image, it just seems the releases .zip file for arm64 may not be valid.

```
$ unzip aws-runas-3.1.2-darwin-arm64.zip 
Archive:  aws-runas-3.1.2-darwin-arm64.zip

$ file aws-runas
./aws-runas: Mach-O 64-bit executable arm64

$ ./aws-runas
Killed: 9

$ ./aws-runas -D
Killed: 9

$ ./aws-runas -v
Killed: 9
```

**Comment by bjbishop** [comment]
this is what my mac platform says:

```
$ uname -a
Darwin ....... 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:20 PDT 2021; root:xnu-7195.141.6~3/RELEASE_ARM64_T8101 arm64
```

**Comment by mmmorris1975** [comment]
There's a couple of issues at play here.  You must be the 1st (congrats, I think?) person to actually use the M1 binary ... or at least the 1st to raise an issue for it.

For some reason the mac builds from CircleCI are a bit janky, it spits out invalid intel binaries too, but I can recompile on my local machine and re-upload to the GH release page and it seems to work well.  I've been doing the same with the arm64 macos releases too, I figured if the intel builds were bad, the arm ones probably weren't working either.  Odd thing is the same cross-compile steps inside CircelCI work fine for Windows, and Linux (intel and arm).

The biggest blocker for me is that I don't have any M1-based system to actually compile or test on, so I have to rely on cross-compiling.  CircleCI provides access to macos VMs, but only for intel processors.  I could try requesting access to those and see if it makes the macos builds better behaved, but other than that, I don't have a lot of great options.

**Comment by bjbishop** [comment]
Thanks!  Yep I'm certainly blazing new trails here with this M1 Mac, so far so good though!

Makes sense, and let me know if I can help with anything, I would be happy to contribute compiled binaries for arm64 from my Mac and send them to you in some way as well.  For now, though compiling from source works just fine.

Thanks again Mike, let me know if I can be of assistance!

**Comment by gmarkey** [comment]
`go install github.com/mmmorris1975/aws-runas@3.1.2` works on M1 running Monterey (golang 1.17.2).

**Comment by mmmorris1975** [comment]
Thanks for the feedback @gmarkey

@bjbishop Looks like you're running Big Sur? (Kernel version jives with my Intel MBP running Big Sur)

**Comment by et304383** [comment]
Chiming in as another user affected!

Curse Apple for moving away from Intel.  Nothing but problems!

**Comment by mmmorris1975** [comment]
With the release of [3.3.0](https://github.com/mmmorris1975/aws-runas/releases/tag/3.3.0), I've built the Mac binaries on an AMD mbp running Monterey (12.2.1), using go 1.17.6.  Since I don't have an ARM Mac, anyone willing to try the ARM64 build out and report back?

**Comment by bjbishop** [comment]
@mmmorris1975 it works!  This was from Big Sur 11.6.4 (20G417) on a 2020 MBP 13" with an M1

```sh
 ~/Downloads % unzip aws-runas-3.3.0-darwin-arm64.zip 
Archive:  aws-runas-3.3.0-darwin-arm64.zip
  inflating: aws-runas               

 ~/Downloads % file aws-runas
aws-runas: Mach-O 64-bit executable arm64

 ~/Downloads % ./aws-runas
2022/02/21 15:43:54 ERROR nothing to do!
NAME:
   aws-runas - Create an environment for interacting with the AWS API using an assumed role

USAGE:
   aws-runas [global options] [subcommand] profile [arguments...]

VERSION:
   3.3.0
```

**Comment by vinodnsshivaram** [comment]
looks like this issue is back again after 3.3.0 (3.3.3, 3.3.2 and 3.3.1 arm64 do not work ) works if i use amd64 
Device: 2021 16inch MacBook Pro - Apple M1 Pro

below errors when using 3.3.2

file aws-runas
aws-runas: Mach-O 64-bit executable arm64

./aws-runas
zsh: killed     ./aws-runas

./aws-runas -D
zsh: killed     ./aws-runas -D

./aws-runas -v
zsh: killed     ./aws-runas -v

uname -a
Darwin ........ 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64

**Comment by mmmorris1975** [comment]
Closed (see #84)
